pkg/graveler/graveler_test.go

```diff
@@ -352,6 +356,122 @@ func TestGraveler_Set(t *testing.T) {
 		})
 	}
 }
+
+func TestGravelerSet_Advanced(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+	committedMgr := &testutil.CommittedFake{}
+	newSetVal := &graveler.ValueRecord{Key: []byte("key"), Value: &graveler.Value{Data: []byte("newValue"), Identity: []byte("newIdentity")}}
+	// RefManager mock base setup
+	refMgr := mock.NewMockRefManager(ctrl)
+	var getFlow = func() {
+		refMgr.EXPECT().ParseRef(gomock.Any()).Times(1).Return(graveler.RawRef{BaseRef: ""}, nil)
+		refMgr.EXPECT().ResolveRawRef(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(&graveler.ResolvedRef{
+			Type:                   0,
+			ResolvedBranchModifier: 0,
+			BranchRecord: graveler.BranchRecord{
+				BranchID: "",
+				Branch: &graveler.Branch{
+					CommitID:     "",
+					StagingToken: "",
+					SealedTokens: nil,
+				},
+			},
+		}, nil)
+		refMgr.EXPECT().GetCommit(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(nil, graveler.ErrCommitNotFound)
+	}
```

Comment:
Suggest to simplify the code with having refExpect and assign the function to name with better explain what it does. (not sure it is the best name)

```suggestion
refExpect := refMgr.EXPECT()
	refExpectCommitNotFound := func() {
		refExpect.ParseRef(gomock.Any()).Times(1).Return(graveler.RawRef{BaseRef: ""}, nil)
		refExpect.ResolveRawRef(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(&graveler.ResolvedRef{
			BranchRecord: graveler.BranchRecord{
				Branch: &graveler.Branch{},
			},
		}, nil)
		refExpect.GetCommit(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(nil, graveler.ErrCommitNotFound)
	}
```